### PR TITLE
feat: Close Conversation Front-end

### PR DESCRIPTION
**Pull request by wadedvsa**
## Description

This ticket is to allow the case worker to close the conversation and the user can no longer send a reply.
The command handler should close the existing conversation and mark the records on the database as closed.
The message will appear on the message tab as a closed conversation

Related issue: [VOL-4689](https://dvsa.atlassian.net/browse/VOL-4689)

## Before submitting (or marking as "ready for review")

- [ ] Does the pull request title follow the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification?
- [ ] Have you performed a self-review of the code
- [ ] Have you have added tests that prove the fix or feature is effective and working
- [ ] Did you make sure to update any documentation relating to this change?

### CODE REVIEW

File: module/Olcs/src/Controller/Messages/LicenceCloseConversationController.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+        $this->flashMessengerHelperService = $flashMessengerHelperService;
+    }
+
+    /**
+     * @return ViewModel|Response
+     */
+    public function confirmAction()
+    {
+        $form = $this->getForm('CloseConversation');
```
Comment (fibble) [inline module/Olcs/src/Controller/Messages/LicenceCloseConversationController.php:45]: We should use the FQCN for the Close Conversation::Class form class here.

**Review by fibble — CHANGES_REQUESTED** [review]

File: module/Olcs/src/Controller/Messages/LicenceCloseConversationController.php (hunk 1 above)
Comment (wadedvsa) [reply module/Olcs/src/Controller/Messages/LicenceCloseConversationController.php]: When I searched for usage the first one I saw was for strings, didn't notice use of FQCN which is definitely better so gladly!

File: module/Olcs/src/Controller/Messages/LicenceCloseConversationController.php [hunk 2]
```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Olcs\Controller\Messages;
+
+use     Common\Controller\Interfaces\ToggleAwareInterface;
```
Comment (fibble) [inline module/Olcs/src/Controller/Messages/LicenceCloseConversationController.php:7]: v.minor whitespace issue here.

**Review by fibble — DISMISSED** [review]
One v.minor comment on whitespace. Otherwise looks good.

File: module/Olcs/src/Controller/Messages/LicenceCloseConversationController.php (hunk 2 above)
Comment (wadedvsa) [reply module/Olcs/src/Controller/Messages/LicenceCloseConversationController.php]: You say minor, I say instantly drove me crazy.

File: module/Olcs/src/Form/Model/Fieldset/CloseConversationText.php [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+use Laminas\Form\Annotation as Form;
+
+/**
+ * @Form\Name("main")
+ */
+class CloseConversationText
+{
+    /**
+     * @Form\Type("\Common\Form\Elements\Types\PlainText")
+     * @Form\Attributes({"value":"The conversation will be removed from the Inbox and a transcript will be archived in docs and attachments tab."})
```
Comment (hobbyhacker0) [inline module/Olcs/src/Form/Model/Fieldset/CloseConversationText.php:16]: This is probably fine as-is, but wondering if it's worth making this text editable? Not even sure how we do that with form field labels

File: module/Olcs/src/Form/Model/Fieldset/CloseConversationActions.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+/**
+ * @codeCoverageIgnore Auto-generated file with no methods
+ * @Form\Name("main")
+ */
+class CloseConversationActions
+{
+    /**
+     * @Form\Attributes({
+     *     "type":"submit",
+     *     "data-module": "govuk-button",
```
Comment (hobbyhacker0) [inline module/Olcs/src/Form/Model/Fieldset/CloseConversationActions.php:18]: Add space before "submit"

File: module/Olcs/src/Form/Model/Fieldset/CloseConversationActions.php [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+     *     "label": "End and archive conversation"
+     * })
+     * @Form\Type("\Common\Form\Elements\InputFilters\ActionButton")
+     */
+    public $close = null;
+
+    /**
+     * @Form\Attributes({
+     *     "data-module": "govuk-button",
+     *     "type": "submit",
```
Comment (hobbyhacker0) [inline module/Olcs/src/Form/Model/Fieldset/CloseConversationActions.php:32]: 2 buttons with type submit might be an accessibility issue

Comment (wadedvsa) [reply module/Olcs/src/Form/Model/Fieldset/CloseConversationActions.php:32]: All situations with a cancel/close and submit button I've found so far all use type submit with and id of cancel: there's a lot!

File: module/Olcs/src/Form/Model/Fieldset/CloseConversationActions.php (hunk 4 above)
Comment (wadedvsa) [reply module/Olcs/src/Form/Model/Fieldset/CloseConversationActions.php:18]: Done that plus tidied up all annotations.

File: module/Olcs/src/Form/Model/Fieldset/CloseConversationActions.php (hunk 5 above)
Comment (wadedvsa) [reply module/Olcs/src/Form/Model/Fieldset/CloseConversationActions.php]: Changed to cancel but the FE forces it to submit - one to work out why!

File: module/Olcs/src/Form/Model/Fieldset/CloseConversationText.php (hunk 3 above)
Comment (wadedvsa) [reply module/Olcs/src/Form/Model/Fieldset/CloseConversationText.php]: As discussed, will look to do this later.

**Review by hobbyhacker0 — APPROVED** [review]

**Review by fibble — APPROVED** [review]

**Pull request merged**